regipy/plugins/system/network_data.py

```diff
@@ -21,80 +20,146 @@ def get_network_info(self, subkey, interfaces=None):
+                                entries["dhcp_lease_terminates_time"] = (
+                                    lease_terminates_time_str
+                                )
+                        except Exception as e:
+                            logger.error(
+                                f"Error retrieving DHCP lease terminates time for interface {interface.name}: {e}"
+                            )
+                            entries["incomplete_data"] = True
+
+                    else:
+                        entries.update(
+                            {
+                                "ip_address": interface.get_value("IPAddress"),
+                                "subnet_mask": interface.get_value("SubnetMask"),
+                                "default_gateway": interface.get_value(
+                                    "DefaultGateway"
+                                ),
+                                "name_server": interface.get_value("NameServer"),
+                                "domain": interface.get_value("Domain"),
+                            }
+                        )
+                except Exception as e:
+                    logger.error(
+                        f"Error processing DHCP/static IP information for interface {interface.name}: {e}"
+                    )
+                    entries["incomplete_data"] = True
+
+                try:
+                    if interface.subkey_count:
+                        sub_interfaces = []
+                        sub_interfaces = self.get_network_info(
+                            interface, sub_interfaces
+                        )
+                        entries["sub_interface"] = sub_interfaces
+                except Exception as e:
+                    logger.error(
+                        f"Error processing sub-interfaces for interface {interface.name}: {e}"
+                    )
+                    entries["incomplete_data"] = True
+
+                interfaces.append(entries)
+
+        except Exception as e:
+            logger.error(f"Error iterating over subkeys in {subkey.path}: {e}")
 
         return interfaces
 
     def run(self):
         self.entries = {}
 
-        for control_set_interfaces_path in self.registry_hive.get_control_sets(
-            INTERFACES_PATH
-        ):
-            try:
-                subkey = self.registry_hive.get_key(control_set_interfaces_path)
-            except RegistryKeyNotFoundException as ex:
-                logger.error(ex)
-                continue
-
-            self.entries[control_set_interfaces_path] = {
-                "timestamp": convert_wintime(
-                    subkey.header.last_modified, as_json=self.as_json
-                )
-            }
-            interfaces = []
-            interfaces = self.get_network_info(subkey, interfaces)
-            self.entries[control_set_interfaces_path]["interfaces"] = interfaces
+        try:
```

Comment:
No need for the high level exception handling if handling all internal expected exceptions.